mailpoet/lib/Automation/Engine/Data/WorkflowRunLog.php

```diff
@@ -0,0 +1,174 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Data;
+
+use DateTimeImmutable;
+use InvalidArgumentException;
+use MailPoet\Automation\Engine\Exceptions\InvalidStateException;
+use MailPoet\Automation\Engine\Exceptions\UnexpectedValueException;
+use MailPoet\Automation\Engine\Utils\Json;
+use Throwable;
+
+class WorkflowRunLog {
+
+  const STATUS_RUNNING = 'running';
+  const STATUS_COMPLETED = 'completed';
+  const STATUS_FAILED = 'failed';
+
+  /** @var int */
+  private $id;
+
+  /** @var int */
+  private $workflowRunId;
+
+  /** @var DateTimeImmutable */
+  private $startedAt;
+
+  /** @var DateTimeImmutable|null */
+  private $completedAt;
+
+  /** @var string */
+  private $status;
+
+  /** @var array */
+  private $error;
+
+  /** @var array */
+  private $data;
+
+  /** @var string */
+  private $stepId;
+
+  /** @var array */
+  private $args;
+
+  public function __construct(
+    int $workflowRunId,
+    string $stepId,
+    array $args,
+    int $id = null
+  ) {
+    $this->workflowRunId = $workflowRunId;
+    $this->stepId = $stepId;
+    $this->args = $args;
+    $this->status = self::STATUS_RUNNING;
```

Comment:
Hmm, now I see that some kind of a "default" state might be needed. So we may as well keep this one.